util/dvsim/Deploy.py

```diff
@@ -287,15 +289,28 @@ def get_timeout_mins(self):
         """Returns the timeout in minutes."""
         return 0
 
-    def extract_runtimes(self, log_text: List):
-        """Input the logfile in lines and extract the job_runtime."""
+    def extract_info_from_log(self, log_text: List):
+        """Extracts information pertaining to the job from its log.
+
+        This method parses the log text after the job has completed, for the
+        extraction of information pertaining to the job's performance. This
+        base class method extracts the job's runtime (i.e. the wall clock time)
+        as reported by the tool. The tool reported runtime is the most accurate
+        since it is devoid of the delays incurred due to infrastructure and
+        setup overhead.
+
+        The extended classes may override this method to extract other pieces
+        of information from the log.
+
+        `log_text` is the job's log file contents as a list of lines.
+        """
         try:
-            self.job_runtime.time, self.job_runtime.unit = get_job_runtime(
-                log_text, self.sim_cfg.tool)
-        except NotImplementedError:
-            log.debug("Using dvsim default runtime due to unsupported tool: " + self.sim_cfg.tool)
-        except SyntaxError:
-            log.debug("Parsing job runtime has syntax error with " + self.sim_cfg.tool)
+            time, unit = get_job_runtime(log_text, self.sim_cfg.tool)
+            self.job_runtime.set(time, unit)
+        except RuntimeError as e:
```

Comment:
s/SyntaxError/RuntimeError. 
NotImplementedError is a derivative of RuntimeError. So this exception handling logic handles both exceptions in one shot.